### PR TITLE
Online resize of cinder volumes

### PR DESCRIPTION
**Pull request by Scsabiii**

### CODE REVIEW

File: cinder/volume/drivers/vmware/volumeops.py [hunk 1]
```diff
@@ -622,6 +631,54 @@ def extend_virtual_disk(self, requested_size_in_gb, path, dc_ref,
...
+        disk = None
+        devices = self._session.invoke_api(vim_util,
+                                           'get_object_property',
+                                           self._session.vim,
+                                           vm_ref,
+                                           'config.hardware.device')
+        for device in devices['VirtualDevice']:
+            if device.__class__.__name__ == "VirtualDisk" and \
+                    device.backing.fileName == path:
+                disk = device
```
Comment (joker-official) [inline cinder/volume/drivers/vmware/volumeops.py]: You could `break` here and then do the `else` from below in an `else` of the `for`.

File: cinder/volume/drivers/vmware/volumeops.py [hunk 2]
```diff
@@ -622,6 +631,54 @@ def extend_virtual_disk(self, requested_size_in_gb, path, dc_ref,
...
+            task = self._session.invoke_api(self._session.vim,
+                                            "ReconfigVM_Task",
+                                            vm_ref,
+                                            spec=config_spec)
+            self._session.wait_for_task(task)
+            LOG.info("Successfully extended virtual disk: %(path)s to "
+                     "%(size)s GB.",
+                     {'path': path, 'size': requested_size_in_gb})
+        else:
+            msg = str.format("Error during online-resize of disk: %(path)s to "
```
Comment (joker-official) [inline cinder/volume/drivers/vmware/volumeops.py]: Please specify what kind of error.

**Comment by joker-official** [comment]
Could you add a config switch to enable online resize, in case we have to disable it?

File: cinder/volume/drivers/vmware/vmdk.py [hunk 3]
```diff
@@ -1746,7 +1749,8 @@ def extend_volume(self, volume, new_size):
             LOG.info("There is no backing for volume: %s; no need to "
                      "extend the virtual disk.", vol_name)
             return
-        if self._in_use(volume) and not volume['multiattach']:
+        if (self._in_use(volume) and not volume['multiattach'] and
+            self.configuration.vmware_online_resize):
```
Comment (joker-official) [inline cinder/volume/drivers/vmware/vmdk.py]: please indent this line one step more to distinguish it from the `if`'s body.

**Comment by kayrus** [comment]
@Scsabiii there is a built-in github feature to squash the PR
Screenshot_20191113_121156

**Pull request merged**